### ISSUE TITLE
Don't include common.gypi from node source tree

### ISSUE THREAD

**Issue by bnoordhuis** [nodejs/node-gyp#1118]
node-gyp includes `$nodedir/common.gypi` when creating build files for an add-on.

Per https://github.com/nodejs/CTC/issues/62#issuecomment-280108504, that probably isn't future-proof and it's a bad idea in general: it makes add-ons pick up compiler and linker settings that aren't intended for them and may be actively harmful.  People have filed bug reports about it.

Plan of action:

1. Merge `common.gypi` into `addon.gypi`.
2. Strip everything that isn't needed by add-ons.
3. Test the bejebus out of it with citgm.

**Comment by murgatroid99** [comment]
I would support this change. I got burned by the fact that `common.gypi` explicitly disables `-Werror`

**Comment by bnoordhuis** [comment]
Yep, but that flows on down from V8.  We'd still carry that change with a separate common.gypi.

**Comment by rvagg** [comment]
still planning on making this happen @bnoordhuis? I fear this is just going to linger as a perpetually open issue unless you make it happen, I suspect there's too many dragons for most mortals to want to attempt this

**Comment by bnoordhuis** [comment]
@rvagg It's one of those projects that I do want to get around to someday but I don't know when. If you want to close it, that's fine by me.

Good job on the cleanup action, by the way. :-)

**Comment by rvagg** [comment]
I think I got throttled along the way with my cleanup, I stopped being able to close issues, a red error box from GitHub as if I was blocked. Seems to be working again now.

I don't mind this being left open if it's genuinely a TODO, your call. I'm just aiming to get rid of the clutter so this starts being manageable again.
